libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker-input.directive.ts

```diff
@@ -173,6 +179,14 @@ export class SkyColorpickerInputDirective
 
   public ngOnInit(): void {
     const element = this.#elementRef.nativeElement;
+    let id = element.id;
+
+    if (!id) {
```

Comment:
This checks if an `id` attribute is set on initialization, but what should happen if the `id` is set by the consumer a few seconds later? Will the colorpicker label still point to the correct ID on the input element?
```
<sky-colorpicker>
  <input [id]="asyncId"
</sky-colorpicker>

// ID is null on init, which triggers the ID Service to create an ID.
protected id = string | null;

// The input's ID is overwritten one second later; does the input service know about it?
setTimeout(() => {
  this.id = 'foobar';
}, 1000);
```